src/theory/arith/nonlinear_extension.cpp

```diff
@@ -3360,6 +3360,53 @@ std::vector<Node> NonlinearExtension::checkTangentPlanes() {
                       << "Tangent plane lemma : " << tlem << std::endl;
                   lemmas.push_back(tlem);
                 }
+
+                // tangent plane reverse implication
+
+                // t <= tplane -> ( (a <= a_v ^ b >= b_v) v (a >= a_v ^ b <=
+                // b_v) ) in clause form, the above becomes t <= tplane -> a <=
+                // a_v v b <= b_v t <= tplane -> b >= b_v v a >= a_v
+                Node a_leq_av = NodeManager::currentNM()->mkNode(LEQ, a, a_v);
+                Node b_leq_bv = NodeManager::currentNM()->mkNode(LEQ, b, b_v);
+                Node a_geq_av = NodeManager::currentNM()->mkNode(GEQ, a, a_v);
+                Node b_geq_bv = NodeManager::currentNM()->mkNode(GEQ, b, b_v);
+
+                Node t_leq_tplane =
+                    NodeManager::currentNM()->mkNode(LEQ, t, tplane);
+                Node a_leq_av_or_b_leq_bv =
+                    NodeManager::currentNM()->mkNode(OR, a_leq_av, b_leq_bv);
+                Node b_geq_bv_or_a_geq_av =
+                    NodeManager::currentNM()->mkNode(OR, b_geq_bv, a_geq_av);
+                Node ub_reverse1 = NodeManager::currentNM()->mkNode(
+                    OR, t_leq_tplane.negate(), a_leq_av_or_b_leq_bv);
+                Trace("nl-ext-tplanes")
+                    << "Tangent plane lemma : " << ub_reverse1 << std::endl;
+                lemmas.push_back(ub_reverse1);
+                Node ub_reverse2 = NodeManager::currentNM()->mkNode(
+                    OR, t_leq_tplane.negate(), b_geq_bv_or_a_geq_av);
+                Trace("nl-ext-tplanes")
+                    << "Tangent plane lemma : " << ub_reverse2 << std::endl;
```

Comment:
Can you change this trace message to something that distinguishes it, like "Tangent plane lemma (reverse) : ".